lib/paquito/types.rb

```diff
@@ -206,6 +206,11 @@ def register(factory, types)
         end
       end
 
+      def recursive?(type)
+        type_attributes = TYPES.fetch(type.name)
+        !recursive_callback?(type_attributes[:packer]) && !recursive_callback?(type_attributes[:unpacker])
```

Comment:
```suggestion
        recursive_callback?(type_attributes[:packer]) || recursive_callback?(type_attributes[:unpacker])
```

Reply:
Benchmark results may improve with this change since I think you were running with a pool size double what was actually needed.